tests/core/math/test_vector2.h

```diff
@@ -366,12 +373,20 @@ TEST_CASE("[Vector2] Plane methods") {
 	CHECK_MESSAGE(
 			vector.reflect(vector_normal).is_equal_approx(Vector2(2.85851197982345523329, -2.197477931904161412358)),
 			"Vector2 reflect with normal should return expected value.");
+
 	CHECK_MESSAGE(
 			vector.project(vector_y) == Vector2(0, 3.4),
 			"Vector2 projected on the Y axis should only give the Y component.");
 	CHECK_MESSAGE(
 			vector.project(vector_normal).is_equal_approx(Vector2(2.0292559899117276166, 0.60126103404791929382)),
 			"Vector2 projected on a normal should return expected value.");
+	CHECK_MESSAGE(
+			vector.project(vector_zero) == Vector2(),
+			"Vector2 projected on zero should return finite.");
+	CHECK_MESSAGE(
+			vector.project(vector_tiny) == Vector2(),
+			"Vector2 projected on a very small normal should return finite.");
```

Comment:
Maybe these should check that the result is finite, instead of checking for what the results are known to be?